### PR TITLE
Adds MediaIdentifier to acces media on device

### PR DESCRIPTION
**Pull request by hplos**
### Description
fixes [IMediaFile doesn't include the FullPath](https://github.com/dimonovdd/Xamarin.MediaGallery/discussions/37)

### API Changes
added a string to IMediaFile.

### CODE REVIEW

File: MediaGallery/MediaFile/MediaFile.android.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ internal MediaFile(string fileName, Uri uri)
             NameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
             Extension = Path.GetExtension(fileName);
             ContentType = MimeTypeMap.Singleton.GetMimeTypeFromExtension(Extension);
+            MediaIdentifier = uri.ToString();
```
Comment (dimonovdd) [inline MediaGallery/MediaFile/MediaFile.android.cs:17]: How can we use this? Have you tried opening a file using this URI?

File: MediaGallery/MediaGallery/PickMedia.ios.cs [hunk 2]
```diff
@@ -145,9 +147,9 @@ public override void DidFinishPicking(PHPickerViewController picker, PHPickerRes
 
             static IEnumerable<IMediaFile> ConvertPickerResults(PHPickerResult[] results)
                 => results
-                .Select(res => res.ItemProvider)
+                .Select(res => new { res.ItemProvider, res.AssetIdentifier })
```
Comment (dimonovdd) [inline MediaGallery/MediaGallery/PickMedia.ios.cs:150]: How will this be handled for older iOS versions?

**Review by dimonovdd — COMMENTED** [review]
I don't quite understand how we can use it. Could you add samples?

File: MediaGallery/MediaFile/MediaFile.android.cs (hunk 1 above)
Comment (hplos) [reply MediaGallery/MediaFile/MediaFile.android.cs:17]: No I have not. The uri.ToString() was already used. I need to spend some time to check what is needed for Android.

File: MediaGallery/MediaGallery/PickMedia.ios.cs (hunk 2 above)
Comment (hplos) [reply MediaGallery/MediaGallery/PickMedia.ios.cs]: I commit an update to solve the anonymous type.

**Comment by Painwraith** [comment]
Could this be used to determine to file size and duration of videos selected?

**Comment by hplos** [comment]
> Could this be used to determine to file size and duration of videos selected?

Yes by requesting the asset using the MediaIdentyfier all information about the asset can be requested.

Adding the MediaIdentyfier to the picked media elements doesn't require acces to the library, but requesting the asset using the media-identifier does.

**Comment by dimonovdd** [comment]
@hplos Do you have any improvements? Or can we close it?

**Comment by dimonovdd** [comment]
@hplos If you decide to continue we will open it again

**Pull request closed without merging**